tools/autograd/gen_variable_type.py

```diff
@@ -91,7 +91,8 @@
     'reflection_pad1d_backward', 'reflection_pad2d_backward',
     'replication_pad1d', 'replication_pad2d', 'replication_pad3d',
     'replication_pad1d_backward', 'replication_pad2d_backward', 'replication_pad3d_backward',
-    'diag', 'masked_scatter', 'masked_select', 'index_fill', 'trace', 'polar'
+    'diag', 'masked_scatter', 'masked_select', 'index_fill', 'trace', 'polar',
+    'cumsum', 'cumprod', 'prod',
```

Comment:
there's an open PR for cumsum so I think we can exclude the cumsum related changes for this PR https://github.com/pytorch/pytorch/pull/53240

Reply:
There is no complex prod backward without cumsum complex support. Let' see who's PR gets landed first and then one of us could merge?

Reply:
OH, I see that `cumsum` does not need any modification when it comes to autograd. Agree, we could skip it here.